impl/src/metadata.rs

```diff
@@ -63,30 +74,182 @@ impl MetadataFetcher for CargoMetadataFetcher {
     assert!(files.toml_path.is_file());
     assert!(files.lock_path_opt.as_ref().map_or(true, |p| p.is_file()));
 
-    // Copy files into a temp directory
+    let tempdir = TempDir::new()?;
+
     // UNWRAP: Guarded by function assertion
-    let cargo_tempdir = {
-      let dir = TempDir::new()?;
-
-      let dir_path = dir.path();
-      let new_toml_path = dir_path.join(files.toml_path.file_name().unwrap());
-      fs::copy(files.toml_path.as_path(), new_toml_path)?;
-      if let Some(lock_path) = files.lock_path_opt.as_ref() {
-        let new_lock_path = dir_path.join(lock_path.file_name().unwrap());
-        fs::copy(lock_path.as_path(), new_lock_path)?;
-      }
-
-      dir
+    let cargo_dir = match self.use_tempdir {
+      false => files.toml_path.as_path().parent().unwrap(),
+      true => {
+        let dir_path = tempdir.path();
+        let new_toml_path = dir_path.join(files.toml_path.file_name().unwrap());
+        fs::copy(files.toml_path.as_path(), new_toml_path)?;
+        if let Some(lock_path) = files.lock_path_opt.as_ref() {
+          let new_lock_path = dir_path.join(lock_path.file_name().unwrap());
+          fs::copy(lock_path.as_path(), new_lock_path)?;
+        }
+
+        dir_path
+      },
     };
 
     MetadataCommand::new()
       .cargo_path(&self.cargo_bin_path)
-      .current_dir(cargo_tempdir.path())
+      .current_dir(cargo_dir)
       .exec()
       .map_err(|e| e.into())
   }
 }
 
+fn clone_crate_src(registry_url: &str, dir: &Path, name: &str, version: &str) -> Result<PathBuf> {
+  let url = url::Url::parse(registry_url)?;
+
+  log::debug!("Cloning binary dependency: {}", &name);
+  let mut cloner = cargo_clone::Cloner::new();
+  cloner
+    .set_registry_url(
+      format!(
+        "{}://{}",
+        url.scheme(),
+        url.host_str().ok_or(anyhow!(
+          "A host string is expected in the settings.registry url"
+        ))?
+      )
+      .as_str(),
+    )
+    .set_out_dir(dir);
+
+  cloner.clone(
+    cargo_clone::CloneMethodKind::Crate,
+    name,
+    Some(version),
+    &Vec::new(),
+  )?;
+
+  let crate_dir = dir.join(format!("{}-{}", name, version));
+  if !crate_dir.exists() {
+    return Err(anyhow!("Directory does not exist"));
+  }
+
+  Ok(crate_dir)
+}
+
+/** 
+ * Ensures a lockfile is generated for a crate on disk
+ *
+ * If a lockfile for the current crate exists in the `raze` lockfiles output directory, that
+ * lockfile will instead be installed into the crate's directory. If no lockfile exists, one
+ * will be generated.
+ */
+fn generate_lockfile(
```

Comment:
Nit: `cargo_generate_lockfile`

Make clear that this is shelling out to cargo.